src/main/java/codeu/controller/BotServlet.java

```diff
@@ -0,0 +1,148 @@
+
+        if (user == null) {
+            response.sendRedirect("/login");
+            return;
+        }
+
+        final String conversationTitle = "chatbot-" + user.hashCode();
+
+        Conversation conversation = conversationStore.getConversationWithTitle(conversationTitle);
+        if (conversation == null) {
+            System.out.println("BotServlet-doGet: Chatbot conversation was null for user `" + user + "`: " + conversationTitle);
+            response.sendRedirect("/conversations");
+            return;
+        }
+
+        UUID conversationId = conversation.getId();
+
+        List<Message> messages = messageStore.getMessagesInConversation(conversationId);
+
+        request.setAttribute("conversation", conversation);
+        request.setAttribute("messages", messages);
+        request.setAttribute("messagePostUrl", "/bot");
+        request.setAttribute("customChatTitle", "EastBot");
+        request.getRequestDispatcher("/WEB-INF/view/chat.jsp").forward(request, response);
+    }
+
+    /**
+     * This function fires when a user submits the form on the chatbot page. It gets the logged-in
+     * username from the session, the chat message from the submitted form data.
+     * It creates a new Message from that data, adds it to the model, and then
+     * redirects back to the chatbot page.
+     */
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response)
+            throws IOException, ServletException {
+
+        String username = (String) request.getSession().getAttribute("user");
+        if (username == null) {
+            // user is not logged in, don't let them add a message
+            response.sendRedirect("/login");
+            return;
+        }
+
+        final String conversationTitle = "chatbot-" + username.hashCode();
+
+        Conversation conversation = conversationStore.getConversationWithTitle(conversationTitle);
+        if (conversation == null) {
+            System.out.println("BotServlet-doPost: Chatbot conversation was null for user `" + username + "`: " + conversationTitle);
+            response.sendRedirect("/conversations");
+            return;
+        }
+
+        User user = userStore.getUser(username);
+        if (user == null) {
+            System.out.println("BotServlet-doPost: No user with username `" + username + "`");
+            response.sendRedirect("/login");
+            return;
+        }
+
+
+        String messageContent = request.getParameter("message");
+
+        // this removes any HTML from the message content
+        String cleanedMessageContent = Jsoup.clean(messageContent, Whitelist.none());
+
+        Message message =
+                new Message(
+                        UUID.randomUUID(),
+                        conversation.getId(),
+                        user.getId(),
+                        cleanedMessageContent,
+                        Instant.now());
+
+        messageStore.addMessage(message);
```

Comment:
For the MVP, we should should forward the message or its content to a service that will take care of doing the NLP, and generating the required response. Make another message out of that response, label it as issued by the bot, and add that to the message store as well c:

Reply:
Makes sense. Would this be difficult to implement?

Reply:
Certainly suitable for an MVP c: